packages/mirror-tv-next/components/tag/more-posts-list.tsx

```diff
@@ -0,0 +1,65 @@
+'use client'
+import UiLoadMreButton from '../shared/ui-load-more-button'
+import { PostByTagName } from '~/graphql/query/posts'
+import { useState } from 'react'
+import { fetchMoreItems } from '~/components/tag/action'
+import errors from '@twreporter/errors'
+import styles from '~/styles/components/tag/more-posts-list.module.scss'
+import PostsList from './posts-list'
+
+type UiMoreTagListProps = {
+  tagName: string
+  pageSize: number
+  postsCount: number
+}
+
+export default function MorePostsList({
+  tagName,
+  pageSize,
+  postsCount,
+}: UiMoreTagListProps) {
+  const [page, setPage] = useState(1)
+  const [postsList, setPostsList] = useState<PostByTagName[]>([])
+
+  const handleClickLoadMore = async () => {
```

Comment:
這個 function 應該只要寫這樣就夠了：
```
 const handleClickLoadMore = async () => {
    const newPosts = await fetchMoreItems(page, tagName, pageSize)
    if (!newPosts) return
    setPage((page) => page + 1)
    setPostsList((oldPost) => [...oldPost, ...newPosts])
  }
```
不需要再包一層 try catch block，fetch data 的 server action 裡已經有了。

Reply:
好奇問一下：如果沒打到資料或是回傳 404 or 500，也不需要 try catch 嗎？（被錯誤處理搞到迷路）

Reply:
思索了一下，應該加這個
```
 if (!newPosts) {
    // This will activate the closest `error.js` Error Boundary
    throw new Error('Failed to fetch data')
  }
```
  這樣會觸發距離最近的 error.tsx

Reply:
如果要在 tag page 內部做錯誤處理，會需要有 posts 無法正確顯示的 UI ，當 tag page folder 的 `error.tsx `file 被觸發，就會顯示該頁面的錯誤UI，而不是直接進入 400 500 error 頁面。
https://nextjs.org/docs/app/building-your-application/routing/error-handling#how-errorjs-works
<img width="668" alt="Screen Shot 2024-01-29 at 1 12 21 PM" src="https://github.com/mirror-tv/Tabris/assets/66411169/16732061-fed5-4843-b2c7-8c6d9481be13">